sdk/identity/identity/review/identity.api.md

```diff
@@ -266,25 +274,22 @@ export interface ManagedIdentityCredentialClientIdOptions extends TokenCredentia
     clientId?: string;
 }
 
-// @public
-export type ManagedIdentityCredentialOptions = ManagedIdentityCredentialClientIdOptions | ManagedIdentityCredentialResourceIdOptions;
-
 // @public
 export interface ManagedIdentityCredentialResourceIdOptions extends TokenCredentialOptions {
     resourceId?: string;
```

Comment:
```suggestion
    resourceId: string;
```

It's fine to leave `clientId` in ClientIdOptions optional, but making this one required, I think, will make it so that the type system treats options that specify neither as ClientId options (which is what the program does at runtime, to my understanding.

Reply:
should I do the same for defaultAzure credential resource options too?